apis/v1alpha2/validation/gateway.go

```diff
@@ -26,6 +27,19 @@ import (
 	gatewayv1a2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
 )
 
+var (
+	exists          = struct{}{}
+	tcpUdpProtocols = map[gatewayv1a2.ProtocolType]struct{}{
+		gatewayv1a2.TCPProtocolType: exists,
+		gatewayv1a2.UDPProtocolType: exists,
+	}
+	nonTlsProtocols = map[gatewayv1a2.ProtocolType]struct{}{
```

Comment:
Nit: idiomatic Go capitalizes initialisms, so Udp and Tls should be UDP and TLS.

https://github.com/golang/go/wiki/CodeReviewComments#initialisms

Reply:
@jpeach I understand but make caps for tcpUdpProtocols to TCPUDPProtocols would export it and I think camel case would be better since we append both the protocols in the name. another var nonTlsProtocols could be made to nonTLSProtocols but then one name would be exported and other would not be so it would be a bit inconsistent .

Reply:
I think the appropriate capitalization is `tcpUDPProtocols` and `nonTLSProtcols`, does that line up with what you were thinking @jpeach?

Reply:
nonTLSProtcols is good but `tcpUDPProtocols` is a bit strange, is there any words we can use to represent TCP and UDP protocols?

Reply:
Maybe we should focus on why we care about these sets of protocols:

- `nonTLSProtcols` -> `protocolsTLSInvalid`
- `tcpUDPProtocols` -> `protocolsHostnameInvalid`

Along with that change, it would probably be helpful to have a brief comment for each explaining the purpose of each set.